docs/C2SMGuidelines.rst

```diff
@@ -0,0 +1,175 @@
+libraries and executables.
+
+Building 
+^^^^^^^^
+There are two possible ways of building software with Spack.
+*spack install* and  *spack dev-build*.
+Both of them are fine, but have some specialties one needs to take
+into account.
+
+Spack install (C2SM Guidelines)
+-------------------------------
+Every *spack-install* needs a version suffix, i.e *spack install <package>@<version-suffix>*.
+This version-suffix can have different meanings:
+
+* branch in the git repository
+* one out of several git repositories defined in the package
+* a specifc version (git-tag) with a corresponding git-hash hardcoded in the package
+
+Only for the last item in the list above you will always fetch and
+compile the same code.  The two other items can lead to different
+codes in case the *HEAD* of this specific branch/repository got some
+additional commits in the meantime.
+
+Especially for production it is very important to now which version of a code is actually used.
+
+The variety of different version-suffix for the cosmo-package:
+
+.. code-block:: python
+
+    git      = 'git@github.com:COSMO-ORG/cosmo.git'
+    apngit   = 'git@github.com:MeteoSwiss-APN/cosmo.git'
+    c2smgit  = 'git@github.com:C2SM-RCM/cosmo.git'
+
+    version('org-master', branch='master', get_full_repo=True)
+    version('apn-mch', git=apngit, branch='mch', get_full_repo=True)
+    version('c2sm-master', git=c2smgit, branch='master', get_full_repo=True)
+    version('c2sm-features', git=c2smgit, branch='c2sm-features', get_full_repo=True)
+    version(5.09, git=c2sm, tag=5.09, get_full_repo=True)
+
+There are three different git repositories available for the cosmo-package:
+
+* COSMO-ORG/cosmo.git: version-suffix *org-master*
+* MeteoSwiss-APN/cosmo.git: version-suffix *apn-mch*
+* C2SM-RCM/cosmo.git: version-suffix *c2sm-master*
+* C2SM-RCM/cosmo.git: version-suffix *c2sm-features* 
+
+It is clear that only using *spack install <package>@5.09* will
+always result in the same code, all other version only point to a
+*HEAD* of a git branch.
+
+The list of versions, including tagged versions, is provided by *spack
+info package_name*. Note that tagged versions for COSMO on the
+*c2sm-features* branch are not yet provided but will be offered
+soon. We thus recommend using the *spack devbuildcosmo* command for
+now.
+
+So long story short:
+
+**Always use a valid git tag as a version-suffix when building
+software with** *spack install* **for production!**
+
+Spack dev-build (C2SM Guidelines)
+---------------------------------
+
+In order to install software with *spack dev-build* one needs a
+local source code.  Spack will compile the code as it is locally
+present. Contrary to *spack install*, version-suffix does not have
+any effect on the code version compiled. Of course the version-suffix
+will appear in the installation path and the Spack database later on
+and will need to be present in so-called *SPEC*.
+
+One could therefore in principle use whatever version name with
```

Comment:
The point here is that with dev-build one can only have a single installation of COSMO, because the version-suffix dev-build can only be used once.

What Matthieu describes here is a way to manually copy the exes back into the sources (i.e. on scratch) where the dev-build command was launched.

This would allow "multiple" instances of a dev-build in parallel.